src/gsy_e/models/strategy/heat_pump.py

```diff
@@ -41,20 +41,26 @@ class HeatPumpStrategy(TradingStrategyBase):
 
     # pylint: disable=too-many-arguments)
     def __init__(self,
-                 maximum_power_rating: float = ConstSettings.HeatPumpSettings.MAX_POWER_RATING_KW,
-                 min_temperatur_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C,
-                 max_temperatur_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C,
-                 external_temperatur_C: float = ConstSettings.HeatPumpSettings.EXT_TEMP_C,
+                 maximum_power_rating_kW: float =
+                 ConstSettings.HeatPumpSettings.MAX_POWER_RATING_KW,
+                 min_temp_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C,
+                 max_temp_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C,
+                 initial_temp_C: float = ConstSettings.HeatPumpSettings.INIT_TEMP_C,
+                 external_temp_C: Optional[float] = None,
+                 external_temp_profile_uuid: Optional[str] = None,
                  tank_volume_l: float = ConstSettings.HeatPumpSettings.TANK_VOL_L,
-                 average_consumption: float =
-                 ConstSettings.HeatPumpSettings.CONSUMPTION_KW,
+                 consumption_kW: Optional[float] = None,
+                 consumption_profile_uuid: Optional[str] = None,
                  source_type: int = ConstSettings.HeatPumpSettings.SOURCE_TYPE,
                  order_updater_parameters: Dict[
                      AvailableMarketTypes, HeatPumpOrderUpdaterParameters] = None,
                  preferred_buying_rate: float =
                  ConstSettings.HeatPumpSettings.PREFERRED_BUYING_RATE
                  ):
 
+        if ConstSettings.MASettings.MARKET_TYPE == 1:
+            raise NotImplementedError("Heatpump has not been implemented for the OneSidedMarket")
```

Comment:
I would even add an assertion here, to denote that this is really something that should not happen under any circumstances.

Reply:
[2f5a617](https://github.com/gridsingularity/gsy-e/pull/1581/commits/2f5a6176aa55f7138056dc02d8793c2d17bd3b17)

[ce93cae](https://github.com/gridsingularity/gsy-e/pull/1581/commits/ce93caed8217088b8346a705f0a79634a47ed8c4)